src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs

```diff
@@ -140,11 +140,14 @@ protected virtual string GetRuntimeSettings(GcMode gcMode, IResolver resolver)
         [PublicAPI]
         protected virtual FileInfo GetProjectFilePath(Type benchmarkTarget, ILogger logger)
         {
-            if (!GetSolutionRootDirectory(out var solutionRootDirectory))
+            if (!GetSolutionRootDirectory(out var rootDirectory))
```

Comment:
nit: to reduce the nesting, could you please join the both conditions using `&&` ?

```cs
if (!GetSolutionRootDirectory(out var rootDirectory) && !GetProjectRootDirectory(out rootDirectory))
```